.github/workflows/ci.yml

```
@@ -37,53 +83,50 @@ jobs:
           - { mock-maker: 'mock-maker-subclass', member-accessor: 'member-accessor-reflection' }
           - { mock-maker: 'mock-maker-inline', member-accessor: 'member-accessor-reflection' }
 
-    # All build steps
-    # SINGLE-MATRIX-JOB means that the step does not need to be executed on every job in the matrix
+    # All test steps
     steps:
+      - name: 1. Checkout out test code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
 
-    - name: 1. Check out code
-      uses: actions/checkout@v3 # https://github.com/actions/checkout
-      with:
-        fetch-depth: '0' # https://github.com/shipkit/shipkit-changelog#fetch-depth-on-ci
-
-    - name: 2. Set up Java ${{ matrix.java }}
-      uses: actions/setup-java@v3
-      with:
-        distribution: 'zulu'
-        java-version: ${{ matrix.java }}
-
-    - name: 3. Validate Gradle wrapper
-      if: matrix.java == 11 && matrix.entry.mock-maker == 'mock-maker-default' # SINGLE-MATRIX-JOB
-      uses: gradle/wrapper-validation-action@v1.0.6 # https://github.com/gradle/wrapper-validation-action
-
-    - name: 4. Build and check reproducibility of artifacts (single job only)
-      if: matrix.java == 11 && matrix.entry.mock-maker == 'mock-maker-default' # SINGLE-MATRIX-JOB
-      run: ./check_reproducibility.sh
-
-    - name: 5. Spotless check (single job only). Run './gradlew spotlessApply' locally if this job fails.
-      if: matrix.java == 11 && matrix.entry.mock-maker == 'mock-maker-default' # SINGLE-MATRIX-JOB
-      run: ./gradlew spotlessCheck
+      - name: 2. Download java 11 build artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: java-11-build-artifacts
 
-    - name: 6. Build on Java ${{ matrix.java }} with ${{ matrix.entry.mock-maker }} and ${{ matrix.entry.member-accessor }}
-      run: ./gradlew build idea --scan
-      env:
-        MOCK_MAKER: ${{ matrix.entry.mock-maker }}
-        MEMBER_ACCESSOR: ${{ matrix.entry.member-accessor }}
+      - name: 3. Set up Java ${{ matrix.java }}
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: ${{ matrix.java }}
+
+      - name: 4. Test on Java ${{ matrix.java }} with ${{ matrix.entry.mock-maker }} and ${{ matrix.entry.member-accessor }}
+        run: |
+          ./gradlew test \
+          -x compileTestJava -x compileTestGroovy -x compileTestKotlin -x compileTestBundleJava \
+          -x classes -x compileJava -x compileGroovy -x compileKotlin -x compileOtherBundleJava \
+          -x processResources -x processOtherBundleResources \
+          -x preBuild -x preDebugBuild \
+          -x jar
```

Tip: when you have just one command you can do this too:
```suggestion
        run: >
          ./gradlew test
          -x compileTestJava -x compileTestGroovy -x compileTestKotlin -x compileTestBundleJava
          -x classes -x compileJava -x compileGroovy -x compileKotlin -x compileOtherBundleJava
          -x processResources -x processOtherBundleResources
          -x preBuild -x preDebugBuild
          -x jar
```
probably worth keeping the `\` version though for consistency with other `run:` blocks.